### PR TITLE
Bump CNM v1.25.23, v1.26.18, v1.27.12, v1.28.4

### PR DESCRIPTION
**Pull request by lzhecheng**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Bump CNM v1.25.23, v1.26.18, v1.27.12, v1.28.4
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 1]
```diff
@@ -91,13 +91,14 @@ $global:imagesToPull += @(
     "mcr.microsoft.com/oss/kubernetes-csi/secrets-store/driver:v1.3.4",
     "mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.4.1",
     # Azure cloud node manager. Owner: nilo19 (Qi Ni)
-    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.25.15", # for k8s 1.25.x
     "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.25.18", # for k8s 1.25.x
-    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.26.11", # for k8s 1.26.x
+    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.25.23", # for k8s 1.25.x
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:95]: Is v1.26.11 still used for now? If so, I think that it is better to keep the old one. Same for other versions.

Comment (lzhecheng) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:95]: I don't know if the strategy has changed.
1. added a new one:1.26.18
2. keep an old one: 1.26.14
3. remove the oldest one: 1.26.11

Comment (lzhecheng) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:95]: I suppose we have 2 for a major version?

Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:95]: It has followed the rule.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Review by cameronmeissner — CHANGES_REQUESTED** [review]
do we really need to be caching this many versions? 

we've recently been running into VHD disk space issues, thus we're trying to be very strict about what we're caching

**Comment by lzhecheng** [comment]
/close
Updated by another PR.

**Pull request closed without merging**